projects/web-wallet/rpc-protocol.md

```diff
@@ -0,0 +1,432 @@
+  }
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "txId": "00004788e55b9b3fb90aaad2065e99ccf36ccd2fac5ed40ae62eeefd2486e96c",
+  "name": "Bathor",
+  "symbol": "BTR",
+  "amount": 500
+}
+```
+
+
+### 2.1.3 `htr_getUtxos`
+
+The `htr_getUtxos` method will search for `utxos`  given a set of filters
+
+**Parameters**
+
+1. `maxUtxos` - (Defaults to 255) Maximum number of `utxos` to return.
+2. `token` - (Defaults to HTR) Token to filter the `utxos`
+3. `filterAddress` - Address to filter the `utxos`. The address must belong to the initialized wallet
+4. `amountSmallerThan` - (Optional) Maximum limit of `utxo` amount to filter the list. We will only return `utxos` that have the `amount` lower than this value. Integer representation of decimals, i.e. 100 = 1.00.
+5. `amountBiggerThan` - (Optional) Minimum limit of `utxo` amount to filter the list. We will only return `utxos` that have the `amount` higher than this value. Integer representation of decimals, i.e. 100 = 1.00.
+6. `maximumAmount` - (Optional) Limit the maximum total amount to return summing all utxos. Integer representation of decimals, i.e. 100 = 1.00.
+7. `onlyAvailableUtxos` - (Defaults to true) Get only available `utxos`, ignoring locked ones.
+
+**Example Request:**
+
+```json
+{
+  "id": 3,
+  "jsonrpc": "2.0",
+  "method": "htr_getUtxos",
+  "params": {
+	"maxUtxos": 50,
+	"token": "BTR",
+	"amount": 1
+  }
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "total_amount_available": 5,
+  "total_utxos_available": 10,
+  "total_amount_locked": 0,
+  "total_utxos_locked": 0,
+  "utxos": [
+    {
+      "address": "HNnK9wgUVL6Cjzs1K3jpoGgqQTXCqpAnW8",
+      "amount": 1,
+      "tx_id": "00fff7a3c6eb95ec3343bffcfca9a3a0d3e243462ae7de1f200cdd76716140fb",
+      "locked": false,
+      "index": 0
+    }
+  ]
+}
+```
+
+
+### 2.1.4 `htr_signWithAddress`
+
+The `htr_signWithAddress` method requests a signed message using an address' private key that can be verified.
+
+**Parameters**
+
+1. `message` - String containing a message to be signed
+2. `addressIndex` - Address index to sign the message with
+
+**Request:**
+
+```json
+{
+  "id": 3,
+  "jsonrpc": "2.0",
+  "method": "htr_signWithAddress",
+  "params": {
+    "message": "sign-me",
+    "address": 5
+  }
+}
+```
+**Response:**
+
+```json
+{
+  "message": "H2/QA48wJhvbKJH6hWXUJPHK5kUTwx4XAxuZGFzUeSPwLeFBv206lOs8ECHMTV/KilGbFP1p5e52mMGTUDFT7WM="
+}
+```
+
+### 2.1.5 `htr_getBalance`
+
+Different from account-based blockchains where users are used to always using the same address on each transaction, getting the balance on hathor involves going through a list of addresses from the user's wallet and calculating the sum of all utxos.
+
+So in order for a `dApp` to know the balance of a given token for the connected user's wallet, we must have a method to fetch it.
+
+The `htr_getBalance` method fetches the balance for a given token.
+
+**Parameters**
+
+1. `token` - (Defaults to `HTR`) May be specified for a token other than `HTR`
```

Comment:
Should we add an optional `addresses` that will return the balance of a subset of addresses? Also maybe a `address_indexes` where the caller can specify the indexes instead of the addresses.

Reply:
I like the idea, done in 657cbc8a619b5de1cdeaa2ad1ae7ff9971ed0d20

Reply:
Great! Thanks!